File: eHour-persistence/src/main/java/net/rrm/ehour/persistence/approvalstatus/dao/ApprovalStatusDaoHibernateImpl.java
package net.rrm.ehour.persistence.approvalstatus.dao;

import java.util.List;

import org.springframework.stereotype.Repository;

import net.rrm.ehour.data.DateRange;
import net.rrm.ehour.domain.Activity;
import net.rrm.ehour.domain.ApprovalStatus;
import net.rrm.ehour.persistence.dao.AbstractGenericDaoHibernateImpl;

@Repository("approvalStatusDao")
public class ApprovalStatusDaoHibernateImpl extends AbstractGenericDaoHibernateImpl<ApprovalStatus, Integer> implements ApprovalStatusDao {

	protected final static String CACHEREGION = "query.ApprovalStatus";
	
	public ApprovalStatusDaoHibernateImpl() {
		super(ApprovalStatus.class);
	}

	@Override
	public List<ApprovalStatus> findApprovalStatusesForActivity(Activity activity, DateRange dateRange) {
		List<ApprovalStatus> results = null;

		String[] keys = new String[] { "activity", "dateStart", "dateEnd"};
		Object[] params = new Object[] { activity, dateRange.getDateStart(), dateRange.getDateEnd() };
		
		results = findByNamedQueryAndNamedParam("ApprovalStatus.findApprovalStatusesForActivity", keys, params, true, CACHEREGION);
		
		return results;		
	}
}

File: eHour-persistence/src/main/scala/net/rrm/ehour/persistence/approvalstatus/dao/ApprovalStatusDaoHibernateImpl.scala
package net.rrm.ehour.persistence.approvalstatus.dao

import java.util

import net.rrm.ehour.data.DateRange
import net.rrm.ehour.domain.{ApprovalStatus, Customer, User}
import net.rrm.ehour.persistence.dao.AbstractGenericDaoHibernateImpl

class ApprovalStatusDaoHibernateImpl extends AbstractGenericDaoHibernateImpl[Integer, ApprovalStatus](classOf[ApprovalStatus]) with ApprovalStatusDao {
  final val CacheRegion = Some("query.ApprovalStatus")

  override def findApprovalStatusesForUserWorkingForCustomer(user: User, customer: Customer, dateRange: DateRange): util.List[ApprovalStatus] = {
    val keys  = List("user", "customer", "dateStart", "dateEnd")
    val params  = List(user, customer, dateRange.getDateStart, dateRange.getDateEnd)

    findByNamedQuery("ApprovalStatus.findApprovalStatusesForUserWorkingForCustomer", keys, params, CacheRegion)
  }

  override def findApprovalStatusesForCustomers(customers: util.List[Customer], dateRange: DateRange): util.List[ApprovalStatus] = {
    val keys  = List("customers", "dateStart", "dateEnd")
    val params  = List(customers, dateRange.getDateStart, dateRange.getDateEnd)

    findByNamedQuery("ApprovalStatus.findApprovalStatusesForCustomers", keys, params, CacheRegion)
  }
}
